tests/baselines/reference/literals-negative.errors.txt

```diff
@@ -8,7 +8,7 @@ tests/cases/compiler/literals-negative.ts(5,9): error TS2352: Neither type 'numb
     var s = <string>(null);
     var b = <boolean>(n);
             ~~~~~~~~~~~~
-!!! error TS2352: Neither type 'number' nor type 'boolean' is assignable to the other.
+!!! error TS2323: Type 'number' is not comparable to type 'boolean'.

```

Comment:
"comparable" is an odd error to issue when complaining about a type assertion. Can we say "x cannot be converted to y" with the "comparable to" error as the next diagnostic chain?